CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs

```diff
@@ -29,34 +29,63 @@ internal static class ImageLoaderHelper
         internal static Dictionary<string, string> SupportedImageFormats =
             new() { { "Supported formats", "*.jpg;*.jpeg;*.jfif;*.png;*.bmp;*.tiff;*.tif;*.webp" } };
 
+        #region Waifu2X
         private static Waifu2X _waifu2X;
+        private static Waifu2XStatus _cachedStatus = Waifu2XStatus.NotInitialized;
 
-        public static Waifu2XStatus Waifu2XStatus => _waifu2X?.Status ?? Waifu2XStatus.NotAvailable;
+        public static Waifu2XStatus Waifu2XStatus
+        {
+            get
+            {
+                // Cache the status of waifu2x
+                if (_cachedStatus == Waifu2XStatus.NotInitialized)
+                {
+                    using var test = CreateWaifu2X();
+                    {
+                        _cachedStatus = test.Status;
+                    }
+                }
+                return _cachedStatus;
+            }
+        }
 
         public static bool IsWaifu2XEnabled
         {
             get => GetAppConfigValue("EnableWaifu2X").ToBool() && IsWaifu2XUsable;
```

Comment:
I think we don't need to check if waifu2x is usable if the config says its disabled

Reply:
> I think we don't need to check if waifu2x is usable if the config says its disabled

a short-circuit evaluation happens here

Reply:
after re reading the docs i finally get it 
:derp:

Thanks for the clarification